src/pages/taskQueue/TaskQueueTable/index.tsx

```diff
@@ -0,0 +1,145 @@
+import { useMemo, useRef } from "react";
+import styled from "@emotion/styled";
+import Badge from "@leafygreen-ui/badge";
+import { LGColumnDef, useLeafyGreenTable } from "@leafygreen-ui/table";
+import { Body, Disclaimer } from "@leafygreen-ui/typography";
+import { useTaskQueueAnalytics } from "analytics";
+import { StyledRouterLink, WordBreak } from "components/styles";
+import { BaseTable } from "components/Table/BaseTable";
+import {
+  getVersionRoute,
+  getTaskRoute,
+  getUserPatchesRoute,
+} from "constants/routes";
+import { TaskQueueItem, TaskQueueItemType } from "gql/generated/types";
+import { formatZeroIndexForDisplay } from "utils/numbers";
+import { msToDuration } from "utils/string";
+
+interface TaskQueueTableProps {
+  taskQueue: Omit<TaskQueueItem, "revision">[];
+  loading: boolean;
+}
+
+const TaskQueueTable: React.FC<TaskQueueTableProps> = ({
+  loading,
+  taskQueue,
+}) => {
+  const taskQueueAnalytics = useTaskQueueAnalytics();
+  const tableContainerRef = useRef<HTMLDivElement>(null);
+
+  const columns = useMemo(
+    () => taskQueueTableColumns(taskQueueAnalytics.sendEvent),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  );
+  const table = useLeafyGreenTable<Omit<TaskQueueItem, "revision">>({
+    data: taskQueue,
+    columns,
+    containerRef: tableContainerRef,
+    enableColumnFilters: false,
+  });
+
+  return (
+    <BaseTable
+      data-cy="task-queue-table"
+      table={table}
+      loading={loading && taskQueue?.length === 0}
+      shouldAlternateRowColor
+    />
+  );
+};
+
+const TaskCell = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+type TaskQueueColumnData = Omit<TaskQueueItem, "revision">;
+
+const taskQueueTableColumns = (
+  sendEvent: ReturnType<typeof useTaskQueueAnalytics>["sendEvent"]
+) => {
+  const columns: LGColumnDef<TaskQueueColumnData>[] = [
+    {
+      header: "",
+      cell: ({ row }) => (
+        <Body weight="medium">{formatZeroIndexForDisplay(row.index)}</Body>
+      ),
+      align: "center",
+      id: "index",
+    },
+    {
+      header: "Task",
+      accessorKey: "displayName",
+      cell: (value) => {
+        const { buildVariant, displayName, id, project } = value.row.original;
+        return (
+          <TaskCell>
+            <Body>
+              <StyledRouterLink
+                data-cy="current-task-link"
+                to={getTaskRoute(id)}
+                onClick={() => sendEvent({ name: "Click Task Link" })}
+              >
+                <WordBreak>{displayName}</WordBreak>
+              </StyledRouterLink>
+            </Body>
```

Comment:
Any reason for the `<Body>` tag here?
```suggestion
              <StyledRouterLink
                data-cy="current-task-link"
                to={getTaskRoute(id)}
                onClick={() => sendEvent({ name: "Click Task Link" })}
              >
                <WordBreak>{displayName}</WordBreak>
              </StyledRouterLink>
```

Reply:
No reason at all apart from some copy pasta :)